src/db.c

```diff
@@ -946,14 +946,15 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
         } while (cursor &&
               maxiterations-- &&
               listLength(keys) < (unsigned long)count);
-    } else if (o->type == OBJ_SET && o->encoding == OBJ_ENCODING_INTSET) {
-        int pos = 0;
-        int64_t ll;
-
-        while(intsetGet(o->ptr,pos++,&ll))
-            listAddNodeTail(keys,createStringObjectFromLongLong(ll));
```

Comment:
now that i think of it, we're loosing the ability of using integer encoded strings and shared integers.
i suppose that's not bad since they'll be converted to strings when written into the output buffer very soon. but maybe the bigger memcpy is an issue?

maybe use setTypeNext instead of setTypeNextObject to mitigate that, and keep the possibility of using createStringObjectFromLongLong?

Reply:
The following 3 tests:
The reason why `createObject(OBJ_STRING,sdsfromlonglong(vll));` is slower than `createStringObject(vstr, vlen);` is that the latter loses the EMBSTR ability.

```c
# cost usec: 195925
for (int i = 0; i < 10000000; i++) {
    vstr = lpGet(p,&vlen,intbuf);
    robj *o = createStringObject(vstr, vlen);
    decrRefCount(o);
}
```

```c
# cost usec: 311655
for (int i = 0; i < 10000000; i++) {
    vstr = lpGetValue(p,&slen,&vll);
    robj *o = createObject(OBJ_STRING,sdsfromlonglong(vll));
    decrRefCount(o);
}
```

```c
# cost usec:: 117145
for (int i = 0; i < 10000000; i++) {
        vstr = lpGetValue(p,&slen,&vll);
        robj *o = createStringObjectFromLongLong(vll);
        decrRefCount(o);
}
```

Reply:
1 memcpy + 2 mallocs, vs 1 memcpy + 1 malloc.
but still createStringObjectFromLongLong is just 1 malloc and 0 memcpy.
and either way, writing it to the output buffer adds a memcpy.

so if we use setTypeNext we'll use either createStringObjectFromLongLong or createStringObject and we can enjoy both embedded string and integer encoded one.